update/process/afterHardFork.go

```diff
@@ -0,0 +1,99 @@
+package process
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/update"
+)
+
+// TODO: do factory which creates the needed components
+// TODO: consensus on the first created block -> integrate the hardfork after processor into the normal block processor ?
+// TODO: save blocks to storage, marshalize, broadcast.
+// TODO: set previous roothash to genesis block
+// TODO: use new genesis block process
+
+// ArgsAfterHardFork defines the arguments for the new after hard fork process handler
+type ArgsAfterHardFork struct {
+	MapBlockProcessors map[uint32]update.HardForkBlockProcessor
+	ImportHandler      update.ImportHandler
+	ShardCoordinator   sharding.Coordinator
+	Hasher             hashing.Hasher
+	Marshalizer        marshal.Marshalizer
+}
+
+type afterHardFork struct {
+	mapBlockProcessors map[uint32]update.HardForkBlockProcessor
+	importHandler      update.ImportHandler
+	shardCoordinator   sharding.Coordinator
+	hasher             hashing.Hasher
+	marshalizer        marshal.Marshalizer
+}
+
+// NewAfterHardForkBlockCreation creates the after hard fork block creator process handler
+func NewAfterHardForkBlockCreation(args ArgsAfterHardFork) (*afterHardFork, error) {
+	if args.MapBlockProcessors != nil {
```

Comment:
== nil instead != nil for all these checks and also use check.IfNil where is possible

Reply:
done